pkg/apis/ndm/v1alpha1/types.go

```diff
@@ -71,3 +71,47 @@ type StoragePoolList struct {
 
 	Items []StoragePool `json:"items"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=disk
+
+// Disk describes disk resource.
+type Disk struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata, omitempty"`
+
+	Spec DiskSpec `json:"spec"`
+}
+
+// DiskSpec is the specification for the disk stored as CRD
+type DiskSpec struct {
+	Uuid       string `json:"uuid"`         //unique identifier for the disk
+	Path       string `json:"path"`         //disk path (e.g. /dev/sdb)
+	Capacity DiskCapacity `json:"capacity"` //capacity (e.g. size, used)
+	Details DiskDetails `json:"details"`    //disk details (e.g. model, serial)
+}
+
+type DiskCapacity struct {
+	Storage   uint64 `json:"storage"`       //disk size (e.g. 200G)
```

Comment:
The comment should also be updated, secondly if we name it bytes it makes more sense perhaps? i.e DiskCapacity.bytes = XXX ? and then we can even make it DiskCapacity.HumanReadble = 20G or whatever in later versions of the API?